checks/raw/pinned_dependencies.go

```diff
@@ -187,10 +189,21 @@ func isDockerfile(pathfn string, content []byte) bool {
 }
 
 func collectDockerfilePinning(c *checker.CheckRequest, r *checker.PinningDependenciesData) error {
-	return fileparser.OnMatchingFileContentDo(c.RepoClient, fileparser.PathMatcher{
+	err := fileparser.OnMatchingFileContentDo(c.RepoClient, fileparser.PathMatcher{
 		Pattern:       "*Dockerfile*",
 		CaseSensitive: false,
 	}, validateDockerfilesPinning, r)
+	if err != nil {
+		return err
+	}
+	//nolint:errcheck
+	remediationMetadata, _ := remediation.New(c)
+
+	for i := range r.Dependencies {
+		rr := &r.Dependencies[i]
+		rr.Remediation = generateRemediation(remediationMetadata, rr)
+	}
+	return nil
```

Comment:
1. previously we only generated remediation in the case where dependencies were unpinned. Do we need to do that here? I'm a bit surprised the tests work as is?

2. (optional) do we still need the helper function? we know the type is `DependencyUseTypeDockerfileContainerImage`, so can we call `remediation.CreateDockerfilePinningRemediation` directly?

Reply:
1 fixed in https://github.com/ossf/scorecard/pull/3632/commits/2397c5402cf979c25049fc63083c20ed64153b96.

Reply:
2 fixed in https://github.com/ossf/scorecard/pull/3632/commits/1fcc17a47c6f05bc8095371207258c340dc46069